### ISSUE TITLE
Make GraphQLResult initialisers public

### ISSUE THREAD

**Issue by halonsoluis** [apollographql/apollo-ios#436]
Currently there's no way of creating a `GraphQLResult` due to every initialiser being internal.

```
GraphQLResult<Query.Data>(
            data: data,
            errors: nil,
            source: .cache,
            dependentKeys: nil
        )
```

Fails with message _Cannot convert value of type 'Query.Data?' to expected argument type '_?'_

and 

`try GraphQLResponse(operation: query, body: json).parseResult()`

Fails due to `parseResult()` being internal

Reading the threads it's clear to me that at least one of those methods was public before. 
What should be the proper way to initialise a custom `GraphQLResult`?

**Comment by g00m** [comment]
+1

**Comment by g00m** [comment]
I want to mock the return value of the `resultHandler: OperationResultHandler<Query>?`

**Comment by rcresnik** [comment]
+1

**Comment by halonsoluis** [comment]
Having an initialiser I can create my own responses in a middleware, as, for example, return a valid manually cached result when offline.

**Comment by tsaarva** [comment]
+1

**Comment by dannyhertz** [comment]
+1

**Comment by mcai1** [comment]
+1

**Comment by sebastiannicklas** [comment]
+1

**Comment by designatednerd** [comment]
This was released with 0.11.1!
